test/python/circuit/test_extensions_standard.py

```diff
@@ -1486,5 +1488,52 @@ def test_kwarg_deprecation(self, instr_name, inst_class, n_params, kwarg_map):
         self.assertEqual(cargs, [])
 
 
+@ddt
+class TestStandardEquivalenceLibrary(QiskitTestCase):
+    """Standard Extension Test."""
+
+    @data(
+        HGate, CHGate, IGate, RGate, RXGate, CRXGate, RYGate, CRYGate, RZGate,
+        CRZGate, SGate, SdgGate, CSwapGate, TGate, TdgGate, U1Gate, CU1Gate,
+        U2Gate, U3Gate, CU3Gate, XGate, CXGate, CCXGate, YGate, CYGate,
+        ZGate, CZGate
+    )
+    def test_definition_parameters(self, gate_class):
+        """Verify decompositions from standard equivalence library match definitions."""
+
+        n_params = len(set(signature(gate_class.__init__).parameters) - {'label', 'self'})
+        param_vector = ParameterVector('th', n_params)
+        float_vector = [0.1 * i for i in range(n_params)]
+
+        param_gate = gate_class(*param_vector)
+        float_gate = gate_class(*float_vector)
+
+        param_entry = std_eqlib.get_entry(param_gate)
+        float_entry = std_eqlib.get_entry(float_gate)
+
+        if not param_gate.definition:
+            self.assertEqual(len(param_entry), 0)
+            self.assertEqual(len(float_entry), 0)
+            return
+
+        if gate_class is CXGate:
+            # CXGate currently has a definition in terms of CXGate.
+            self.assertEqual(len(param_entry), 0)
+            self.assertEqual(len(float_entry), 0)
+            return
+
+        self.assertEqual(len(param_entry), 1)
+        self.assertEqual(len(float_entry), 1)
+
+        param_qc = QuantumCircuit(param_gate.num_qubits)
+        float_qc = QuantumCircuit(float_gate.num_qubits)
+
+        param_qc.append(param_gate, param_qc.qregs[0])
+        float_qc.append(float_gate, float_qc.qregs[0])
+
+        self.assertEqual(param_entry[0], param_qc.decompose())
+        self.assertEqual(float_entry[0], float_qc.decompose())
```

Comment:
have to be careful that once you make `qc.decompose()` rely on `StandardEquivalenceLibrary` and not `gate.definition`, this assert becomes a truism - so have to remember to update this test to check actual operator matrix equivalence between LHS and RHS of the entries.